### ISSUE TITLE
Move Che release workflow from Centos CI

### ISSUE THREAD

**Issue by mkuznyetsov** [eclipse-che/che#17857]
### Is your task related to a problem? Please describe.

Centos CI was announced to be deprecated around the end of this autumn. Before that happens, we would want to move to the alternative options, that would allow us to perform release processes. 

This issue is only about Che release related projects, for example, QE test jobs are not covered here.

List of projects, that require CI for release:

- [x] [machine-exec](https://github.com/eclipse/che/issues/18778) (docker image, Golang project)
  - [x] remove cico action che-machine-exec-release
- [x] devfile-registry + plugin-registry (docker image)
  - [x] remove cico actions for release branch
- [x] [che-theia](https://github.com/eclipse/che/issues/18779) (docker images)
  - [x] remove cico action for release branch, che-theia-che-release
- [x] che-dashboard (docker image, Typescript project)
- [x] che-workspace-loader (docker image, Golang project)
- [x] [che-server](https://github.com/eclipse/che/issues/18780) (docker images, Java Maven project + publishing on Nexus, publishing Typescipt DTO on npmjs.org)
- [x] [che-docs](https://github.com/eclipse/che/issues/18823) (simple tagging GH action)
  - [x] remove cico action for release branch, che-docs-che-release
- [x] che-operator (docker image, generating & pushing OLM files)
- [x] [community PRs](https://github.com/eclipse/che/issues/18627) (generated PRs only)
- [x] chectl (node JS build)
- [x] overall che-release orchestrator
  - [x] add PHASES
  - [x] verify work for full set of phases 1-7
- [x] remove cico che-* credentials -- won't do, too many remaning jobs to migrate first - see https://github.com/eclipse/che/issues/19287
- [x] overall che-release readme/docs
  - [x] update che-release issue template: https://github.com/eclipse/che/pull/19063
  - [x] update che-release docs: remove stuff that no longer applies, document new process triggers (GH actions) https://github.com/eclipse/che-release/blob/master/README.md#phase-2---manual-steps

We should also ensure that CI builds (nightly, latest, next), tests, and Hosted Che jobs are also migrated away from Centos CI. 

See https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/resources/jenkins/ci-centos/defaults.yaml

Ideally, we would want to preserve a workflow, of having 1 primary workflow, that would run all the subsequent tasks in the correct order, considering that some builds must be triggered after others, some can be run in parallel etc..  

One of the alternatives to be considered is Github Actions. Some of the projects, e.g. Che Dashboard already run daily builds, so we should try creating a similar job, that would cover full release workflow for it.

**Comment by nickboldt** [comment]
chectl moved to GH action in https://github.com/che-incubator/chectl/runs/1494300249?check_suite_focus=true

**Comment by nickboldt** [comment]
Initial attempt to (easily) move to a GH action to invoke che-release process has failed, see https://github.com/eclipse/che/issues/18576

**Comment by nickboldt** [comment]
plugin and devfile registry RELEASE moved to GH actions:

PR to remove plugin and devfile release actions from cico: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/13061

NOTE: this does not affect the release-preview job that Hosted Che uses https://ci.centos.org/job/devtools-che-plugin-registry-release-preview/ which should ALSO be migrated away from CICO eventually.

**Comment by nickboldt** [comment]
Looks like the che-bot (or the GH action bot?) doesn't have permission to push to `che-incubator/community-operators`.

```
 create mode 100644 upstream-community-operators/eclipse-che/7.23.0/checlusters.org.eclipse.che.crd.yaml
 create mode 100644 upstream-community-operators/eclipse-che/7.23.0/eclipse-che.v7.23.0.clusterserviceversion.yaml

   - Push branch update-eclipse-che-upstream-operator-7.23.0 to the 'che-incubator/community-operators' GitHub repository
Warning: Permanently added the RSA host key for IP address '140.82.114.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Error: Process completed with exit code 128.
```

-- https://github.com/eclipse/che-operator/runs/1557932455?check_suite_focus=true

**Comment by benoitf** [comment]
I've checked that che-bot has write permissions to che-incubator/community-operators github repository

**Comment by benoitf** [comment]
Token need to be given to git tool (as it's an external org/repository)

**Comment by nickboldt** [comment]
@benoitf has created a new bot: https://github.com/che-incubator-bot and shared its credentials with the che-incubator org. 

Latest news on the ci centos shutdown can be seen in https://issues.redhat.com/browse/APPSRE-2140

**Comment by nickboldt** [comment]
Only a few weeks left (one sprint) before cico is decommissioned! 

Do we have all the nightlies, tests, and releases moved over to GH action? cc: @rhopp @benoitf

**Comment by nickboldt** [comment]
@mkuznyetsov I've added a couple TODOs to this issue:

- [ ] update che-release issue template: #19063
- [ ] update che-release docs: remove stuff that no longer applies, document new process triggers (GH actions) https://github.com/eclipse/che-release/blob/master/README.md#phase-2---manual-steps

Please review my PR for the first, and submit a PR for the second.

**Comment by mkuznyetsov** [comment]
and looks like all essential release jobs for upstream Che release are now finally done on GitHub Actions workflows. So it's time for old jobs & docs cleanup.

**Comment by nickboldt** [comment]
Still waiting on a review of https://github.com/eclipse/che/pull/19063 so I can merge it... :( 7 days = sadness

Also waiting on a rebase / fix in https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/15206/

**Comment by benoitf** [comment]
you can ping me before if I forgot some reviews

**Comment by nickboldt** [comment]
getting closer to being done here - both PR and MR from my previous comment are merged.

**Comment by nickboldt** [comment]
We could consider removing these chunks of code:

https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/resources/jenkins/ci-centos/defaults.yaml#L2133 - 2152

https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/resources/jenkins/ci-centos/defaults.yaml#L325 - 406

but not sure if that will break anything that @ibuziuk or @rhopp still need to run there.

**Comment by mkuznyetsov** [comment]
I've removed credentials for che-operator, that I have myself added for operator release on CentOS CI, so I know that only I have used them

But as for `remove cico che-* credentials?` section, I'm not sure if I can do it, since those still may be used by some nigthly/Hosted Che related jobs

**Comment by nickboldt** [comment]
Seems we still have some actively running che jobs on CICO:

![image](https://user-images.githubusercontent.com/227597/109059997-dc776300-76b2-11eb-9a45-4c431ed62d19.png)
![image](https://user-images.githubusercontent.com/227597/109060096-00d33f80-76b3-11eb-8d71-01ff9f775c02.png)

Also seeing lots of recently-run builds here:

https://ci.centos.org/view/Devtools/

**Comment by ericwill** [comment]
The release preview jobs are for Hosted Che, which will EOL in 4 days. After that we can remove them.

**Comment by nickboldt** [comment]
oh, good! what about ...

![image](https://user-images.githubusercontent.com/227597/109060561-8ce56700-76b3-11eb-81a1-0843ca3d4e90.png)

and

![image](https://user-images.githubusercontent.com/227597/109060625-a090cd80-76b3-11eb-83ad-b6a34ac2ee5b.png)

**Comment by nickboldt** [comment]
For example
https://ci.centos.org/view/Devtools/job/devtools-che-devfile-registry-build-master/261/console -> 
https://quay.io/repository/eclipse/che-devfile-registry?tag=ecabf9b&tab=tags seems to be the current process for publishing nightlies.

**Comment by ericwill** [comment]
> For example
> https://ci.centos.org/view/Devtools/job/devtools-che-devfile-registry-build-master/261/console ->
> https://quay.io/repository/eclipse/che-devfile-registry?tag=ecabf9b&tab=tags seems to be the current process for publishing nightlies.
> 
> but there's also https://github.com/eclipse/che-devfile-registry/blob/master/.github/workflows/nightly-build-publish.yml

It's a duplicate job, we've had successful nightly build jobs via GitHub action for ages now:
https://github.com/eclipse/che-devfile-registry/runs/1928193391?check_suite_focus=true

I'll remove it from the cico job list

**Comment by ericwill** [comment]
> > For example
> > https://ci.centos.org/view/Devtools/job/devtools-che-devfile-registry-build-master/261/console ->
> > https://quay.io/repository/eclipse/che-devfile-registry?tag=ecabf9b&tab=tags seems to be the current process for publishing nightlies.
> > but there's also https://github.com/eclipse/che-devfile-registry/blob/master/.github/workflows/nightly-build-publish.yml
> 
> It's a duplicate job, we've had successful nightly build jobs via GitHub action for ages now:
> https://github.com/eclipse/che-devfile-registry/runs/1928193391?check_suite_focus=true
> 
> I'll remove it from the cico job list

It's done: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/15557

**Comment by nickboldt** [comment]
any updates here re: doc?

**Comment by nickboldt** [comment]
Re: remove cico che-* credentials
We won't do this - there are too many remaning jobs to migrate first - see #19287 for followup (which probably needs to be split into prod and qe sub tasks).

**Comment by nickboldt** [comment]
Doc update: 
* https://github.com/eclipse/che-release/pull/32 (merged)

Refactoring/cleanup/add missing DWO/DWCO steps: 
* https://github.com/eclipse/che-release/pull/33
